### ISSUE TITLE
Yolo is unable to load image, fails with exception, undescriptive message

### ISSUE THREAD

**Issue by xumix** [ultralytics/ultralytics#4496]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Training

### Bug

I just started training Yolo on my image classification dataset which contains ~1M images, installed with pip as described in docs.
Running this command:
`yolo classify train model=yolov8m-cls.pt device=0 data=~/images cfg=cfg.yaml batch=1`
Getting this error:

```
Overriding /home/tambovtsev/yolo-pip/lib/python3.10/site-packages/ultralytics/cfg/default.yaml with cfg.yaml
Ultralytics YOLOv8.0.159 🚀 Python-3.10.12 torch-2.0.1+cu118 CUDA:0 (NVIDIA GeForce RTX 3090 Ti, 24248MiB)
engine/trainer: task=classify, mode=train, model=yolov8m-cls.pt, data=/home/tambovtsev/images, epochs=100, patience=50, batch=1, imgsz=500, save=True, save_period=-1, cache=False, device=0, workers=8, project=None, name=None, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, show=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, vid_stride=1, line_width=None, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, boxes=True, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=None, workspace=4, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, label_smoothing=0.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, mosaic=1.0, mixup=0.0, copy_paste=0.0, cfg=None, tracker=botsort.yaml, save_dir=runs/classify/train10
train: /home/tambovtsev/images/train... found 1696589 images in 7704 classes ✅
val: /home/tambovtsev/images/val... found 212478 images in 5177 classes ✅
test: /home/tambovtsev/images/test... found 212485 images in 5137 classes ✅
Overriding model.yaml nc=1000 with nc=7704

                   from  n    params  module                                       arguments
  0                  -1  1      1392  ultralytics.nn.modules.conv.Conv             [3, 48, 3, 2]
  1                  -1  1     41664  ultralytics.nn.modules.conv.Conv             [48, 96, 3, 2]
  2                  -1  2    111360  ultralytics.nn.modules.block.C2f             [96, 96, 2, True]
  3                  -1  1    166272  ultralytics.nn.modules.conv.Conv             [96, 192, 3, 2]
  4                  -1  4    813312  ultralytics.nn.modules.block.C2f             [192, 192, 4, True]
  5                  -1  1    664320  ultralytics.nn.modules.conv.Conv             [192, 384, 3, 2]
  6                  -1  4   3248640  ultralytics.nn.modules.block.C2f             [384, 384, 4, True]
  7                  -1  1   2655744  ultralytics.nn.modules.conv.Conv             [384, 768, 3, 2]
  8                  -1  2   7084032  ultralytics.nn.modules.block.C2f             [768, 768, 2, True]
  9                  -1  1  10854424  ultralytics.nn.modules.head.Classify         [768, 7704]
YOLOv8m-cls summary: 141 layers, 25641160 parameters, 25641160 gradients
Transferred 228/230 items from pretrained weights
AMP: running Automatic Mixed Precision (AMP) checks with YOLOv8n...
AMP: checks passed ✅
WARNING ⚠️ imgsz=[500] must be multiple of max stride 32, updating to [512]
optimizer: SGD(lr=0.01, momentum=0.9) with parameter groups 38 weight(decay=0.0), 39 weight(decay=0.0005), 39 bias(decay=0.0)
Image sizes 512 train, 512 val
Using 0 dataloader workers
Logging results to runs/classify/train10
Starting training for 100 epochs...

      Epoch    GPU_mem       loss  Instances       Size
      1/100     0.843G     0.1402          1        512:   0%|          | 1034/1696589 [00:12<5:38:54, 83.38it/s]
Traceback (most recent call last):
  File "/home/tambovtsev/yolo-pip/bin/yolo", line 8, in <module>
    sys.exit(entrypoint())
  File "/home/tambovtsev/yolo-pip/lib/python3.10/site-packages/ultralytics/cfg/__init__.py", line 432, in entrypoint
    getattr(model, mode)(**overrides)  # default args from model
  File "/home/tambovtsev/yolo-pip/lib/python3.10/site-packages/ultralytics/engine/model.py", line 377, in train
    self.trainer.train()
  File "/home/tambovtsev/yolo-pip/lib/python3.10/site-packages/ultralytics/engine/trainer.py", line 193, in train
    self._do_train(world_size)
  File "/home/tambovtsev/yolo-pip/lib/python3.10/site-packages/ultralytics/engine/trainer.py", line 334, in _do_train
    for i, batch in pbar:
  File "/home/tambovtsev/yolo-pip/lib/python3.10/site-packages/tqdm/std.py", line 1182, in __iter__
    for obj in iterable:
  File "/home/tambovtsev/yolo-pip/lib/python3.10/site-packages/ultralytics/data/build.py", line 38, in __iter__
    yield next(self.iterator)
  File "/home/tambovtsev/yolo-pip/lib/python3.10/site-packages/torch/utils/data/dataloader.py", line 633, in __next__
    data = self._next_data()
  File "/home/tambovtsev/yolo-pip/lib/python3.10/site-packages/torch/utils/data/dataloader.py", line 677, in _next_data
    data = self._dataset_fetcher.fetch(index)  # may raise StopIteration
  File "/home/tambovtsev/yolo-pip/lib/python3.10/site-packages/torch/utils/data/_utils/fetch.py", line 51, in fetch
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/home/tambovtsev/yolo-pip/lib/python3.10/site-packages/torch/utils/data/_utils/fetch.py", line 51, in <listcomp>
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/home/tambovtsev/yolo-pip/lib/python3.10/site-packages/ultralytics/data/dataset.py", line 263, in __getitem__
    sample = self.torch_transforms(im)
  File "/home/tambovtsev/yolo-pip/lib/python3.10/site-packages/torchvision/transforms/transforms.py", line 95, in __call__
    img = t(img)
  File "/home/tambovtsev/yolo-pip/lib/python3.10/site-packages/ultralytics/data/augment.py", line 885, in __call__
    imh, imw = im.shape[:2]
AttributeError: 'NoneType' object has no attribute 'shape'

```
The problem is, I'm unable to find the image causing the error since there are too many of them. 
Could you please add some error checking and debug info if the image fails to load?

### Environment

Ultralytics YOLOv8.0.159 🚀 Python-3.10.12 torch-2.0.1+cu118 CUDA:0 (NVIDIA GeForce RTX 3090 Ti, 24248MiB)

**Comment by glenn-jocher** [comment]
@xumix 👋 hi, thanks for letting us know about this possible problem with YOLO 🚀. We've created a few short guidelines below to help users provide what we need in order to start investigating a possible problem.

## How to create a Minimal, Reproducible Example

When asking a question, people will be better able to provide help if you provide **code** that they can easily understand and use to **reproduce** the problem. This is referred to by community members as creating a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/). Your code that reproduces the problem should be:

- ✅ **Minimal** – Use as little code as possible to produce the problem
- ✅ **Complete** – Provide all parts someone else needs to reproduce the problem
- ✅ **Reproducible** – Test the code you're about to provide to make sure it reproduces the problem

For [Ultralytics](https://ultralytics.com/) to provide assistance your code should also be:

- ✅ **Current** – Verify that your code is up-to-date with GitHub with `git pull` or `pip install -U ultralytics` to ensure your problem has not already been solved in the latest code version.
- ✅ **Unmodified** – Your problem must be reproducible using official Ultrlaytics code without changes. [Ultralytics](https://ultralytics.com/) does not provide support for custom code ⚠️.

If you believe your problem meets all the above criteria, please close this issue and raise a new one using the 🐛 **Bug Report** [template](https://github.com/ultralytics/ultralytics/issues/new/choose) with a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us better understand and diagnose your problem. 

Thank you! 😃

**Comment by xumix** [comment]
![2116397249](https://github.com/ultralytics/ultralytics/assets/250118/90aab5c6-37f2-47fb-b144-54645b38f9b3)
Here is the problem file

**Comment by glenn-jocher** [comment]
@xumix thanks for the image! I'll try to train on this and see what happens.

**Comment by glenn-jocher** [comment]
@xumix I placed this image in coco8/images/train, and started training with the command `yolo train`, and the image is correctly identified and excluded from training. It's likely that you've modified your code or are using out of date or 3rd party changes that are causing your problem.

<img width="1192" alt="Screenshot 2023-08-22 at 23 57 10" src="https://github.com/ultralytics/ultralytics/assets/26833433/04fcb001-3d87-4b18-9260-aa80a7497d53">

**Comment by glenn-jocher** [comment]
@xumix closing as unable to reproduce any problems in official Ultralytics code and issue is likely due to modified or out of date code.

**Comment by glenn-jocher** [comment]
@xumix ah I see you're training classification. This uses a different dataloader. I'm able to reproduce with `yolo train model=yolov8n-cls.pt data=imagenet10` after adding to imagenet10 dataset. Adding a TODO to track this issue for resolution.

**Comment by xumix** [comment]
> @xumix closing as unable to reproduce any problems in official Ultralytics code and issue is likely due to modified or out of date code.

I installed YOLO today on a fresh venv using official YOLO docs, so everything must be up to date. I'm training classification, this may be the key difference

**Comment by glenn-jocher** [comment]
@xumix yes, it looks like we should apply the same dataset cleaning checks to Classify as we do for Detect/Segment/Pose. This is currently done here:

https://github.com/ultralytics/ultralytics/blob/b890e1c937aefa9df2df3cce66d4f0226f4ef42b/ultralytics/data/utils.py#L66-L78

**Comment by glenn-jocher** [comment]
@xumix great news 😃! Your original issue may now be resolved ✅ in PR #4502. This PR adds Classify dataset scanning and caching features, so now Classify training and validation will first verify all images in a dataset are problem-free before training or validating. The results are stored in a *.cache file so that repeated trainings do not re-scan the dataset unless it changes.

To get this update:
- **[Git](https://github.com/ultralytics/ultralytics)** – Run `git pull` from within your `ultralytics/` directory or run `git clone https://github.com/ultralytics/ultralytics` again
- **[Pip](https://pypi.org/project/ultralytics/)** – Update with `pip install -U ultralytics`
- **[Notebooks](https://github.com/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb)** – Check out the updated notebooks  <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/ultralytics)** – Run `sudo docker pull ultralytics/ultralytics:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and letting us know. Please confirm if this update fixes the issue for you, and don't hesitate to report any other **issues** you find or **feature requests** you may have. Happy training with YOLOv8 🚀!

**Comment by glenn-jocher** [comment]
@xumix you're welcome! I'm glad to hear that your issue was resolved. If you encounter any other issues or have further questions during your use of YOLOv8, don't hesitate to reach out. Happy training!
